graph/topo/paton_cycles_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package topo
+
+import (
+	"math"
+	"reflect"
+	"sort"
+	"testing"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/internal/ordered"
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+var undirectedCyclesInTests = []struct {
+	g    []intset
+	want [][][]int64
+}{
+	{
+		g: []intset{
```

Comment:
Would it be worth adding to this graph another component that has a cycle? Like:
```
10: linksTo(11, 12),
11: linksTo(12),
12: nil,
```